src/api/app/services/staging/staging_project_creator.rb

```diff
@@ -0,0 +1,43 @@
+module Staging
+  class StagingProjectCreator
+    def initialize(request_body, staging_workflow)
+      @request_body = request_body
+      @staging_workflow = staging_workflow
+    end
+
+    def call
+      if @request_body.blank?
```

Comment:
That's a neat case of Test Driven Development, but it's hardly the only way to post invalid input.

Reply:
if you added XML validation, you don't need this anymore.